frontend/components/text2text/results/Text2TextList.vue

```diff
@@ -185,15 +184,41 @@ export default {
       itemNumber: 0,
       newSentence: undefined,
       editionMode: false,
+      sentencesOrigin: undefined,
       shiftPressed: false,
       shiftKey: undefined,
       refresh: 1
     };
   },
-  computed: {
-    predictionsLength() {
-      return this.predictions.length;
+  watch: {
+    annotationEnabled(oldValue, newValue) {
+      if (oldValue !== newValue) {
+        // TODO: RESET OR MANTAIN STATE ??
+        this.initializeViewSettings();
+      }
+    },
+    editionMode(oldValue, newValue) {
+      if (oldValue !== newValue) {
+        this.$emit('update-view-config', 'editionMode', this.editionMode)
+      }
+    },
+    itemNumber(oldValue, newValue) {
+      if (oldValue !== newValue) {
+        this.$emit('update-view-config', 'itemNumber', this.itemNumber)
+      }
     },
+    newSentence(oldValue, newValue) {
+      if (oldValue !== newValue) {
+        this.$emit('update-view-config', 'newSentence', this.newSentence)
+      }
+    },
+    sentencesOrigin(oldValue, newValue) {
+      if (oldValue !== newValue) {
+        this.$emit('update-view-config', 'sentencesOrigin', this.sentencesOrigin)
+      }
+    },
+  },
```

Comment:
Could we avoid defining watchers if we use directly the recordViewSettings attributes?